bot/exts/events/advent_of_code/_cog.py

```diff
@@ -172,6 +175,72 @@ async def join_leaderboard(self, ctx: commands.Context) -> None:
         else:
             await ctx.message.add_reaction(Emojis.envelope)
 
+    @in_month(Month.NOVEMBER, Month.DECEMBER)
+    @adventofcode_group.command(
+        name="link",
+        aliases=("connect",),
+        brief="Tie your Discord account with your Advent of Code name."
+    )
+    @whitelist_override(channels=AOC_WHITELIST)
+    async def aoc_link_account(self, ctx: commands.Context, aoc_name: str = None) -> None:
+        """
+        Link your Discord Account to your Advent of Code name.
+
+        Stored in a Redis Cache, Discord ID: Advent of Code Name
+        """
+        cache_items = await self.account_links.items()
+
+        # A short circuit in case the cache is empty
+        if len(cache_items) == 0 and aoc_name:
+            log.info(f"{ctx.author} ({ctx.author.id}) is now linked to {aoc_name}.")
+            await self.account_links.set(ctx.author.id, aoc_name)
+            await ctx.reply(f"You have linked your Discord ID to {aoc_name}.")
+            return
+        elif len(cache_items) == 0:
+            await ctx.reply(
+                "You have not linked an Advent of Code account."
+                "Please re-run the command with one specified."
+            )
+            return
+
+        cache_aoc_name = [value for _, value in cache_items]
+
+        if aoc_name:
+            # Let's check the current values in the cache to make sure it isn't already tied to a different account
+            if aoc_name == await self.account_links.get(ctx.author.id):
+                await ctx.reply(f"{aoc_name} is already tied to your account.")
+                return
+            elif aoc_name in cache_aoc_name:
+                log.info(
+                    f"{ctx.author} ({ctx.author.id}) tried to connect their account to {aoc_name},"
+                    " but it's already connected to another user."
+                )
+                await ctx.reply(
+                    f"{aoc_name} is already tied to another account."
+                    " Please contact an admin if you believe this is an error."
+                )
+                return
+
+            # Update an existing link
+            if old_aoc_name := await self.account_links.get(ctx.author.id):
+                log.info(f"{ctx.author} ({ctx.author.id}) has changed their link from {old_aoc_name} to {aoc_name}.")
+                await self.account_links.set(ctx.author.id, aoc_name)
+                await ctx.reply(f"Your linked account has been changed to {aoc_name}.")
+            else:
+                # Create a new link
+                log.info(f"{ctx.author} ({ctx.author.id}) is now linked to {aoc_name}.")
+                await self.account_links.set(ctx.author.id, aoc_name)
+                await ctx.reply(f"You have linked your Discord ID to {aoc_name}.")
+        else:
```

Comment:
You can probably move the `await self.account_links.set()` line outside(after it) the if..else statement as it is the same.

Reply:
What would be the benefit of doing so here? I want to make sure that the link happens before I send the message back to the user in case something goes wonky. I don't want to tell the user that the link has happened before it's actually happened.

Reply:
No real benefits, would just not duplicate code, looking at the log statements, I think we could even set the new `aoc_name` before the if...else block, since the logging statements are in past, saying the change has already happened.

Reply:
I agree with some of what Jason is saying.

since Since the logging statement is before it happens, it should be in present or future tense.

If I was in a late debugging session trying to figure out what happened, I would think that the account successfully got linked, until I dug a bit further.

Changing the tense would save some time, and make more sense, as being past tense implies it has happened, when it hasn't.